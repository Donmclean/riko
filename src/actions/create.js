import { regexReplaceCustomBoilerplateString, genericLog } from '../utils/functions';
import { cwd } from '../utils/variables';
import fs from 'fs-extra';
import regexReplace from 'regex-replace';

export default async (finalPath, actionType, fileType, fileName) => {
    const fullBinFilePath = `${finalPath}/${fileType}`;

    const filePathStats = await fs.stat(fullBinFilePath);

    //handle Directories
    if(filePathStats.isDirectory()) {
        try {
            await fs.copy(fullBinFilePath, `${cwd}/${fileName}`);
        } catch (err) {
            console.error(`ERROR > error creating directory (${cwd}/${fileName})`, err);
            return 1;
        }

        try {
            await regexReplace('<:rikofilename:>', fileName, `${cwd}/${fileName}`);
        } catch (err) {
            console.error(`ERROR > error replacing custom filenames in (${cwd}/${fileName})`, err);
            return 1;
        }

        genericLog(`${fileName} created successfully`);

    } else { //handle files
        let customizedFile;
        try {
            customizedFile = await fs.readFile(fullBinFilePath, 'utf8');
        } catch (err) {
            console.error(`ERROR > error reading file (${cwd}/${fileName})`, err);
            return 1;
        }

        const editedContent = regexReplaceCustomBoilerplateString(customizedFile, fileName);

        try {
            await fs.writeFile(`${cwd}/${fileName}`, editedContent);
        } catch (err) {
            console.error(`ERROR > error writing to file (${cwd}/${fileName})`, err);
            return 1;
        }

        genericLog(`${fileName} created successfully`);
    }
};