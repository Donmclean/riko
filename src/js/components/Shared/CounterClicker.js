"use strict";
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

//IMPORTANT!: Import actions to dispatch
import getActions from '../../actions/_index';

class CounterClicker extends Component {
    constructor(props) {
        super(props);
        //Get all actions
        this.actions = getActions(this.props.dispatch);
    }

    render() {
        const { state } = this.props;
        return (
            <div>
                <h4>Synchronous Actions</h4>

                <button className="inc" onClick={this.actions.counterActionCreators.shouldIncrement}>INCREMENT</button>
                <button className="dec" onClick={this.actions.counterActionCreators.decrement}>DECREMENT</button>

                <h1 className="counter-val">Counter Reducer State: {state.counterReducer.get('value')}</h1>
            </div>
        );
    }
}

CounterClicker.propTypes = {
    state: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
    return {state};
};

export default connect(mapStateToProps)(CounterClicker);