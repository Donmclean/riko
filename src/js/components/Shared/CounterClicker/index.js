"use strict";
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import { Button } from './counterClickerStyles';

//IMPORTANT!: Import actions to dispatch
import getActions from '../../../_ActionCreators';

class CounterClicker extends Component {
    constructor(props) {
        super(props);
        //Get all actions
        this.actions = getActions(this.props.dispatch);
    }

    render() {
        const { state } = this.props;
        return (
            <div>
                <h4>Synchronous Actions</h4>

                <Button onClick={this.actions.counterActionCreators.shouldIncrement}>INCREMENT</Button>
                <Button onClick={this.actions.counterActionCreators.decrement}>DECREMENT</Button>

                <h1>Counter Reducer State: {state.counterReducer.get('value')}</h1>
            </div>
        );
    }
}

CounterClicker.propTypes = {
    state: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({state});

export default connect(mapStateToProps)(CounterClicker);