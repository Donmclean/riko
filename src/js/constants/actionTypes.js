export const SHOULD_INCREMENT = 'SHOULD_INCREMENT';
export const SHOULD_DECREMENT = 'SHOULD_DECREMENT';
export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT';

export const UPDATE_POSTS = 'UPDATE_POSTS';
export const UPDATE_POST_NUMBER = 'UPDATE_POST_NUMBER';