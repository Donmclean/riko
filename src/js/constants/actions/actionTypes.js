export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT';

export const GET_POSTS = 'GET_POSTS';
export const UPDATE_INPUT_VALUE = 'UPDATE_INPUT_VALUE';