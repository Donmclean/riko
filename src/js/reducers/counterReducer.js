"use strict";
import { Map } from 'immutable';

//Every reducer should have an initial state variable here
const initialState = Map({
    value: 0
});

const counterReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'INCREMENT': {
            return state.update('value', (value) => value + 1);
        }
        case 'DECREMENT': {
            return state.update('value', (value) => value - 1);
        }
        default: {
            break;
        }
    }

    return state;
};

export default counterReducer;