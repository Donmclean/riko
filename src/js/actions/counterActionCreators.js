import * as types from '../../js/constants/actions/actionTypes';

export const shouldIncrement = () => ({
    type: types.SHOULD_INCREMENT
});

export const increment = () => ({
    type: types.INCREMENT
});

export const shouldDecrement = () => ({
    type: types.SHOULD_DECREMENT
});

export const decrement = () => ({
    type: types.DECREMENT
});