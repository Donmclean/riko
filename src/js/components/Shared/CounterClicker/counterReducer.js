"use strict";
import { Map } from 'immutable';
import * as types from '../../../constants/actionTypes';

//Every reducer should have an initial state variable here
export const initialState = Map({ // Export it for testing purposes
    value: 0
});

const counterReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.SHOULD_INCREMENT: {
            break;
        }
        case types.INCREMENT: {
            return state.update('value', (value) => value + 1);
        }
        case types.SHOULD_DECREMENT: {
            break;
        }
        case types.DECREMENT: {
            return state.update('value', (value) => value - 1);
        }
        default: {
            break;
        }
    }

    return state;
};

export default counterReducer;