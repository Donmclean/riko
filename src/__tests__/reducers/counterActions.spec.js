"use strict";

import React from 'react';
import * as counterActions from '../../js/actions/counterActionCreators';
import * as types from '../../js/constants/actions/actionTypes';
import counterReducer, { initialState } from '../../js/reducers/counterReducer';
import { Map } from 'immutable';

//For testing components
import CounterClicker from '../../js/components/shared/CounterClicker';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
const middlewares = []; //eg: [thunk] for async
const mockStore = configureStore(middlewares);

describe('Counter Actions', () => {
    //Initialize mockStore with initial state of Reducer
    const store = mockStore({counterReducer: initialState});

    it('should create a increment action', () => {
        const expectedAction = {
            type: types.INCREMENT
        };
        expect(counterActions.increment()).toEqual(expectedAction);
    });

    it('should create a decrement action', () => {
        const expectedAction = {
            type: types.DECREMENT
        };
        expect(counterActions.decrement()).toEqual(expectedAction);
    });

    it(`${types.INCREMENT} executes successfully --WITH-- IMMUTABLE JS`, () => {
        //Setup
        const expectedState = Map({value: 1});

        //Execute
        const newState = counterReducer(initialState, counterActions.increment());

        //Verify
        expect(newState).toEqual(expectedState);
    });

    it(`${types.DECREMENT} executes successfully --WITH-- IMMUTABLE JS`, () => {
        //Setup
        const expectedState = Map({value: -1});

        //Execute
        const newState = counterReducer(initialState, counterActions.decrement());

        //Verify
        expect(newState).toEqual(expectedState);
    });

    it('renders <CounterClicker /> component successfully in to the DOM with required props', () => {
        //Verify props
        const wrapper = shallow(<CounterClicker store={store}/>);
        expect(wrapper.prop('store')).toEqual(store);

        //Verify Component Snapshot
        const component = renderer.create(
            <CounterClicker store={store}/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});