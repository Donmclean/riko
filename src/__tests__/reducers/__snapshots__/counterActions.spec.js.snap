exports[`Counter Actions renders <CounterClicker /> component successfully in to the DOM with required props 1`] = `
<div>
  <h4>
    Synchronous Actions
  </h4>
  <button
    className="inc"
    onClick={[Function]}>
    INCREMENT
  </button>
  <button
    className="dec"
    onClick={[Function]}>
    DECREMENT
  </button>
  <button>
    Rx INCREMENT
  </button>
  <button>
    Rx DECREMENT
  </button>
  <h1
    className="counter-val">
    Counter Reducer State: 
    0
  </h1>
</div>
`;
